components/TitleBar.tsx

```diff
@@ -0,0 +1,146 @@
+        </div>
+        <SvgUser
+          width={24}
+          height={24}
+          className={`z-[98] mr-4 cursor-pointer text-stone-600`}
+          onClick={() => setShowUserWindow((prevValue) => !prevValue)}
+        />
+        <UserDetailModal
+          showUserWindow={showUserWindow}
+          user={userDetails}
+          closeUserWindow={() => setShowUserWindow(false)}
+        />
+      </div>
+    </div>
+  );
+}
+
+function UserDetailModal({
+  user,
+  showUserWindow,
+  closeUserWindow,
+}: HTMLAttributes<HTMLDivElement> & {
+  user: User;
+  showUserWindow: boolean;
+  closeUserWindow: () => void;
+}) {
+  const router = useRouter();
+
+  const queryClient = useQueryClient();
+
+  const { isPending, error, mutateAsync } = useMutation({
+    mutationKey: QueryKeys.LOGOUT,
+    mutationFn: async () => await axios.post('/api/auth/logout'),
+    onSuccess: () => {
+      queryClient.clear();
+      router.push('/logout').catch((e) => console.error(e));
+    },
+  });
+
+  useShowErrorToast(error);
+
+  if (user && showUserWindow) {
+    return (
+      <>
+        <div
+          className={`fixed left-0 top-0 h-full w-full max-w-full bg-transparent ${isPending ? 'z-[100]' : ''}`}
+          onClick={() => {
+            // this blocks the closing of the User Modal if request for logout is sent
+            if (!isPending) {
+              closeUserWindow();
+            }
+          }}
+        />
+        <div className="absolute right-1 top-12 z-[99] w-56 rounded-md border-2 border-lines bg-bgForms shadow-md shadow-black">
+          <p className="overflow mb-0 ml-3 mt-3 font-bold [overflow-wrap:anywhere]">
+            {user.firstName} {user.lastName}
+          </p>
+          <p className="ml-3 [overflow-wrap:anywhere]">{user.email}</p>
+          <div className="flex w-full flex-col justify-center">
+            <AdminPanelButton user={user} />
+            <Button
+              className="mb-4 ml-3 mr-3 mt-4 flex h-8 w-auto max-w-56 items-center justify-center rounded-md bg-primary text-sm font-medium"
+              onClick={async () => {
+                try {
+                  await mutateAsync();
+                } catch (e) {
+                  handleErrorToast(handleError(e));
+                }
+              }}
+              disabled={isPending}
+            >
+              {' '}
+              Kirjaudu ulos
+              {isPending ? (
+                <SvgSpinner
+                  width={18}
+                  height={18}
+                  className="ml-2 animate-spin"
+                />
+              ) : (
+                <SvgArrowRightStartOnRectangle
+                  width={18}
+                  height={18}
+                  className="ml-2"
+                />
+              )}
+            </Button>
+          </div>
+        </div>
+      </>
+    );
+  }
+  return null;
+}
+
+function AdminPanelButton({ user }: { user: User }) {
+  if (user.role !== 'ADMIN') {
+    return null;
+  }
```

Comment:
Joitain huvittavia juttuja täältä näköjään löytyy, tuonkin voisi lyhentää :smile:  Tässä vain tarkistetaan näytetäänkö käyttäjälle nappula vai ei
```suggestion
function AdminPanelButton(user: User) {
  if (user.role !== 'ADMIN') {
    return null;
  }
```